core/domain/blog_services.py

```diff
@@ -0,0 +1,613 @@
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        blog_post_rights: BlogPostRights or None. Rights object for the given
+            blog post.
+
+    Returns:
+        bool. Whether the given user can edit the given blog post.
+    """
+    if blog_post_rights is None:
+        return False
+    if role_services.ACTION_EDIT_ANY_BLOG_POST in user.actions:
+        return True
+    if blog_post_rights.is_editor(user.user_id):
+        return True
+
+    return False
+
+
+def deassign_user_from_all_blog_posts(user_id):
+    """Removes the user from the list of editor_ids for all blog posts.
+
+    Args:
+        user_id: str. ID to be removed from editor_ids.
+    """
+    blog_models.BlogPostRightsModel.deassign_user_from_all_blog_posts(
+        user_id)
+
+
+def generate_url_fragment(title):
+    """Generates the url fragment for a blog post from the title of the blog
+    post.
+
+    Args:
+        title: str. The title of the blog post.
+
+    Returns:
+        str. The url fragment of the blog post.
+    """
+    lower_title = title.lower()
+    hyphenated_title = lower_title.replace(' ', '-')
+    return hyphenated_title
+
+
+def generate_summary_of_blog_post(content):
+    """Generates the summary for a blog post from the content of the blog
+    post.
+
+    Args:
+        content: santized html str. The blog post content to be truncated.
+
+    Returns:
+        str. The summary of the blog post.
+    """
+    raw_text = html_cleaner.strip_html_tags(content)
+    max_chars_in_summary = constants.MAX_CHARS_IN_BLOG_POST_SUMMARY - 3
+    summary = raw_text[:max_chars_in_summary] + '...'
+    return summary
+
+
+def compute_summary_of_blog_post(blog_post):
+    """Creates BlogPostSummary domain object from BlogPost domain object.
+
+    Args:
+        blog_post: BlogPost. The blog post domain object.
+
+    Returns:
+        BlogPostSummary. The blog post summary domain object.
+    """
+    summary = generate_summary_of_blog_post(blog_post.content)
+
+    return blog_domain.BlogPostSummary(
+        blog_post.id,
+        blog_post.author_id,
+        blog_post.title,
+        summary,
+        blog_post.url_fragment,
+        blog_post.tags,
+        blog_post.thumbnail_filename,
+        blog_post.last_updated,
+        blog_post.published_on)
+
+
+def apply_change_dict(blog_post_id, change_dict):
+    """Applies a changelist to blog post and returns the result.
+
+    Args:
+        blog_post_id: str. ID of the given blog post.
+        change_dict: dict. A dict containing all the changes keyed
+            by corresponding field name (title, content,
+            thumbnail_filename, tags).
+
+    Returns:
+        UpdatedBlogPost. The modified blog post object.
+    """
+    blog_post = get_blog_post_by_id(blog_post_id)
+
+    if 'title' in change_dict:
+        blog_post.update_title(change_dict['title'])
```

Comment:
Do we have any object for the change-dict-schema? [This is going to be used for the schema validation of the args received from the client.]

Reply:
No we dont have, but before updating any value the validation is performed on the argument. Also only 3 fields can be updated: Thumbnail_filename, content and title without requiring any commit mssg or special rights. So simple validation before updating can serve the purpose.